tasks/frontEnd/src/components/ReviewForm.vue

```diff
@@ -0,0 +1,138 @@
+<template>
+  <Form @submit="handleSubmit">
+    <TextField label="Shop name" v-model="review.shopName" autoComplete="off" required :error="shopNameError"></TextField>
+    <TextField label="Product name" v-model="review.productName" autoComplete="off" :error="productNameError"></TextField>
+    <TextField label="Your name" v-model="review.userName" autoComplete="off" :error="userNameError"></TextField>
+    <TextField type="email" label="Your email" v-model="review.userEmail" autoComplete="email" :error="userEmailError"></TextField>
+    <Text>Rating</Text>
+    <InlineStack gap="200">
+      <RadioButton v-for="n in 5" :label="n" :value="n" v-model="review.rating" :checked="review.rating == n" />
+    </InlineStack>
+    <TextField label="Review" v-model="review.reviewBody" :multiline="4" autoComplete="off" :error="reviewBodyError" />
+    <Checkbox label="Verified review" v-model="review.verified" />
+    <DropZone accept="image/*" type="image" @drop="handleDrop" label="Related images">
+      <Layout v-if="review.files.length" vertical>
+        <LayoutSection v-for="file, index in review.files" :key="index">
+            <Thumbnail
+              size="small"
+              :alt="file.name"
+              :source="getSource(file)"
+            />
+            <div>
+              {{ file.name }}{{ ' ' }}
+              <Text variant="bodySm" as="p">
+                {{ file.size }} bytes
+              </Text>
+            </div>
+        </LayoutSection>
+      </Layout>
+      <DropZoneFileUpload v-else actionTitle="Add review images" />
+    </DropZone>
+    <Divider />
+    <Button type="submit" :disabled="disableSubmit">Submit</Button>
+  </Form>
+</template>
+
+<script setup>
+import { reactive, ref, watch } from 'vue';
+
+const emit = defineEmits(['addReview'])
+
+const review = reactive({ shopName: '', productName: '', userName: '', userEmail: '', rating: 3, reviewBody: '', verified: false, files: [] });
+
+const shopNameError = ref(null);
+const productNameError = ref(null);
+const userNameError = ref(null);
+const userEmailError = ref(null);
+const reviewBodyError = ref(null);
+
+const disableSubmit = ref(true);
+
+watch(review, () => {
```

Comment:
I would set these refs to `computed` variables, and remove the need for multiple `watch` functions. Or perhaps it can be refactored another way, thoughts?

Reply:
Yeah agreed on the multiple watch functions especially. This was definitely an area I wanted to refactor if I had more time